### ISSUE TITLE
Protobuf version conflict.

### ISSUE THREAD

**Issue by mapleroyal** [s0md3v/sd-webui-roop#104]
TLDR:
open-clip-torch 2.7.0 requires protobuf==3.20.0
mediapipe 0.10.1 requires protobuf<4,>=3.11
onnx 1.14.0 requires protobuf>=3.20.2

This is on a fresh auto1111 install. I added the following extensions, and have used SD with zero errors until the issue this post is about:
- adetailer
- sd-webui-controlnet
- sd-webui-inpaint-anything

```
(venv) C:\Repos\stable-diffusion-webui>pip install insightface==0.7.3
Collecting insightface==0.7.3
  Downloading insightface-0.7.3.tar.gz (439 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 439.5/439.5 kB 5.5 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: matplotlib in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from insightface==0.7.3) (3.7.1)
Requirement already satisfied: requests in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from insightface==0.7.3) (2.31.0)
Requirement already satisfied: scipy in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from insightface==0.7.3) (1.11.0)
Requirement already satisfied: scikit-image in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from insightface==0.7.3) (0.20.0)
Collecting albumentations
  Downloading albumentations-1.3.1-py3-none-any.whl (125 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 125.7/125.7 kB ? eta 0:00:00
Collecting easydict
  Downloading easydict-1.10.tar.gz (6.4 kB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: numpy in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from insightface==0.7.3) (1.23.5)
Collecting onnx
  Downloading onnx-1.14.0-cp310-cp310-win_amd64.whl (13.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 13.3/13.3 MB 54.4 MB/s eta 0:00:00
Requirement already satisfied: tqdm in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from insightface==0.7.3) (4.65.0)
Requirement already satisfied: Pillow in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from insightface==0.7.3) (9.5.0)
Collecting cython
  Using cached Cython-0.29.35-py2.py3-none-any.whl (988 kB)
Collecting scikit-learn
  Downloading scikit_learn-1.2.2-cp310-cp310-win_amd64.whl (8.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 8.3/8.3 MB 53.2 MB/s eta 0:00:00
Collecting prettytable
  Downloading prettytable-3.8.0-py3-none-any.whl (27 kB)
Collecting opencv-python-headless>=4.1.1
  Downloading opencv_python_headless-4.7.0.72-cp37-abi3-win_amd64.whl (38.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 38.1/38.1 MB 50.4 MB/s eta 0:00:00
Collecting qudida>=0.0.4
  Downloading qudida-0.0.4-py3-none-any.whl (3.5 kB)
Requirement already satisfied: PyYAML in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from albumentations->insightface==0.7.3) (6.0)
Requirement already satisfied: packaging>=20.0 in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from scikit-image->insightface==0.7.3) (23.1)
Requirement already satisfied: PyWavelets>=1.1.1 in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from scikit-image->insightface==0.7.3) (1.4.1)
Requirement already satisfied: lazy_loader>=0.1 in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from scikit-image->insightface==0.7.3) (0.2)
Requirement already satisfied: imageio>=2.4.1 in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from scikit-image->insightface==0.7.3) (2.31.1)
Requirement already satisfied: networkx>=2.8 in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from scikit-image->insightface==0.7.3) (3.1)
Requirement already satisfied: tifffile>=2019.7.26 in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from scikit-image->insightface==0.7.3) (2023.4.12)
Requirement already satisfied: kiwisolver>=1.0.1 in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from matplotlib->insightface==0.7.3) (1.4.4)
Requirement already satisfied: cycler>=0.10 in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from matplotlib->insightface==0.7.3) (0.11.0)
Requirement already satisfied: contourpy>=1.0.1 in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from matplotlib->insightface==0.7.3) (1.1.0)
Requirement already satisfied: python-dateutil>=2.7 in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from matplotlib->insightface==0.7.3) (2.8.2)
Requirement already satisfied: fonttools>=4.22.0 in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from matplotlib->insightface==0.7.3) (4.40.0)
Requirement already satisfied: pyparsing>=2.3.1 in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from matplotlib->insightface==0.7.3) (3.1.0)
Collecting protobuf>=3.20.2
  Using cached protobuf-4.23.3-cp310-abi3-win_amd64.whl (422 kB)
Requirement already satisfied: typing-extensions>=3.6.2.1 in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from onnx->insightface==0.7.3) (4.6.3)
Requirement already satisfied: wcwidth in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from prettytable->insightface==0.7.3) (0.2.6)
Requirement already satisfied: urllib3<3,>=1.21.1 in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from requests->insightface==0.7.3) (1.26.16)
Requirement already satisfied: charset-normalizer<4,>=2 in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from requests->insightface==0.7.3) (3.1.0)
Requirement already satisfied: idna<4,>=2.5 in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from requests->insightface==0.7.3) (3.4)
Requirement already satisfied: certifi>=2017.4.17 in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from requests->insightface==0.7.3) (2023.5.7)
Collecting joblib>=1.1.1
  Downloading joblib-1.2.0-py3-none-any.whl (297 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 298.0/298.0 kB 19.2 MB/s eta 0:00:00
Collecting threadpoolctl>=2.0.0
  Downloading threadpoolctl-3.1.0-py3-none-any.whl (14 kB)
Requirement already satisfied: colorama in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from tqdm->insightface==0.7.3) (0.4.6)
Requirement already satisfied: six>=1.5 in c:\repos\stable-diffusion-webui\venv\lib\site-packages (from python-dateutil>=2.7->matplotlib->insightface==0.7.3) (1.16.0)
Building wheels for collected packages: insightface, easydict
  Building wheel for insightface (pyproject.toml) ... done
  Created wheel for insightface: filename=insightface-0.7.3-cp310-cp310-win_amd64.whl size=841705 sha256=1a39a33952b18db74959869ef539b8db24a54d85794172ac6593104aedda136e
  Stored in directory: c:\users\admin\appdata\local\pip\cache\wheels\e3\d0\80\e3773fb8b6d1cca87ea1d33d9b1f20a223a6493c896da249b5
  Building wheel for easydict (setup.py) ... done
  Created wheel for easydict: filename=easydict-1.10-py3-none-any.whl size=6515 sha256=8118e095b36f40a8a462416e2b1c9ca96c9bfff87827d1d1f7b7683512314fe8
  Stored in directory: c:\users\admin\appdata\local\pip\cache\wheels\b7\e3\8c\f814820ac7ef5fd96f38759ce0d3dc51166a3169008b31da53
Successfully built insightface easydict
Installing collected packages: easydict, threadpoolctl, protobuf, prettytable, opencv-python-headless, joblib, cython, scikit-learn, onnx, qudida, albumentations, insightface
  Attempting uninstall: protobuf
    Found existing installation: protobuf 3.20.0
    Uninstalling protobuf-3.20.0:
      Successfully uninstalled protobuf-3.20.0
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
open-clip-torch 2.7.0 requires protobuf==3.20.0, but you have protobuf 4.23.3 which is incompatible.
mediapipe 0.10.1 requires protobuf<4,>=3.11, but you have protobuf 4.23.3 which is incompatible.
Successfully installed albumentations-1.3.1 cython-0.29.35 easydict-1.10 insightface-0.7.3 joblib-1.2.0 onnx-1.14.0 opencv-python-headless-4.7.0.72 prettytable-3.8.0 protobuf-4.23.3 qudida-0.0.4 scikit-learn-1.2.2 threadpoolctl-3.1.0

[notice] A new release of pip available: 22.2.1 -> 23.1.2
[notice] To update, run: python.exe -m pip install --upgrade pip

(venv) C:\Repos\stable-diffusion-webui>pip install protobuf==3.20.0
Collecting protobuf==3.20.0
  Using cached protobuf-3.20.0-cp310-cp310-win_amd64.whl (903 kB)
Installing collected packages: protobuf
  Attempting uninstall: protobuf
    Found existing installation: protobuf 4.23.3
    Uninstalling protobuf-4.23.3:
      Successfully uninstalled protobuf-4.23.3
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
onnx 1.14.0 requires protobuf>=3.20.2, but you have protobuf 3.20.0 which is incompatible.
Successfully installed protobuf-3.20.0

**Comment by x4080** [comment]
I experience the same, any solution

**Comment by adifelice-godaddy** [comment]
You may have to upgrade `tensorflow`. Try this after activating your venv:
```
pip install tensorflow==2.13.0rc2
```

**Comment by adifelice-godaddy** [comment]
I believe it was closed because dependency issues are not within the scope, as the same `venv` is used by multiple extensions. 

Try to upgrade `open-clip-torch`:
```
pip install open-clip-torch==2.20.0
```

Also, make sure your `webui-user.bat` or equivalent is not reinstalling requirements.

If you get an error about old generated `protobuf` files, then try the `tensorflow` upgrade:
```
pip install tensorflow==2.13.0rc2
```

I hope the above helps.

**Comment by x4080** [comment]
In my case, I revert from pytorch nightly (M2 pro) - it works fine now

**Comment by Buumcode** [comment]
[SOLVED]
In my case (installing Comfyui-reactor-node) I have two errors:
fastapi 0.94.0 requires pydantic!=1.7,!=1.7.1,!=1.7.2,!=1.7.3,!=1.8,!=1.8.1,<2.0.0,>=1.6.2, but you have pydantic 2.8.2 which is incompatible.
open-clip-torch 2.20.0 requires protobuf<4, but you have protobuf 5.27.3 which is incompatible.
-----
For solve first, I ugrade Fastapi version:
1) From root folder > Mouse click on adress bar > Type 'cmd' and press Enter - so it's open command prompt.
.\venv\Scripts\activate
pip install --upgrade fastapi
It's installed newest version of Fastapi – whicj is compatible.
------
For second, I upgrade open-clip-torch:
.\venv\Scripts\activate
pip3 install open-clip-torch
This thing in result make it:
Uninstalling protobuf-5.27.3:
      Successfully uninstalled protobuf-5.27.3
Successfully installed protobuf-3.20.3
------
And all problems was solved.
Hope it's helps.
